gslb/ingestion/bootup_handler.go

```diff
@@ -0,0 +1,198 @@
+	apiServerPort = 9453
+)
+
+func InitializeClusterClient(cfg *restclient.Config) (client.Client, error) {
+	clusterScheme = k8sruntime.NewScheme()
+	utilruntime.Must(clientgoscheme.AddToScheme(clusterScheme))
+	utilruntime.Must(amkov1alpha1.AddToScheme(clusterScheme))
+
+	c, err := client.New(cfg, client.Options{
+		Scheme: clusterScheme,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error in getting client: %v", err)
+	}
+
+	return c, nil
+}
+
+type AMKOClusterReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+func createController() {
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:             clusterScheme,
+		MetricsBindAddress: metricsAddr,
+		Port:               apiServerPort,
+	})
+	if err != nil {
+		gslbutils.Errf("unable to start manager: %v", err)
+		os.Exit(1)
+	}
+
+	if err = (&AMKOClusterReconciler{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		gslbutils.Errf("unable to create cluster reconciler: %v", err)
+		os.Exit(1)
+	}
+	go func() {
+		if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
+			gslbutils.Errf("problem starting manager: %v", err)
+			os.Exit(1)
+		}
+	}()
+}
+
+var currentLeader bool
+
+func acceptGenerationChangePredicate() predicate.Predicate {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			e.ObjectOld.GetGeneration()
+			// skip if no generation change, applicable to all objects
+			return e.ObjectOld.GetGeneration() != e.ObjectNew.GetGeneration()
+		},
+	}
+}
+
+func (r *AMKOClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	gslbutils.Debugf("Starting AMKOCluster reconcialation")
+
+	var amkoClusterList amkov1alpha1.AMKOClusterList
+	err := r.List(ctx, &amkoClusterList, &client.ListOptions{
+		Namespace: federator.AviSystemNS,
+	})
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("AMKOClusterObjects can't be listed, err: %v", err)
+	}
+	if len(amkoClusterList.Items) > 1 {
+		return reconcile.Result{}, fmt.Errorf("only one AMKOClusterObject allowed per cluster")
+	}
+
+	amkoCluster := amkoClusterList.Items[0]
+	if currentLeader != amkoCluster.Spec.IsLeader {
+		gslbutils.Warnf("AMKO leader flag changed, API server would be shut down")
+		apiserver.GetAmkoAPIServer().ShutDown()
+		return ctrl.Result{}, nil
+	}
+
+	currentLeader = amkoCluster.Spec.IsLeader
+	if !currentLeader {
+		gslbutils.Warnf("AMKO is not a leader, will return")
+		return ctrl.Result{}, nil
+	}
+
+	memberClusters, err := federator.FetchMemberClusterContexts(ctx, amkoCluster.DeepCopy())
+	gslbutils.Debugf("memberClusters obtained during reconcialation: %v", memberClusters)
+
+	err = federator.ValidateMemberClusters(ctx, memberClusters)
+	if err != nil {
+		apiserver.GetAmkoAPIServer().ShutDown()
+		return reconcile.Result{}, fmt.Errorf("error in validating the member clusters: %v", err)
+	}
+
+	gslbutils.Debugf("AMKOCluster reconcialation completed")
```

Comment:
I think we should change this to say, `AMKOCluster validation done`.

Reply:
Done